orbit/pkg/table/tcc_access/tcc_access_test.go

```diff
@@ -0,0 +1,77 @@
+//go:build darwin
+// +build darwin
+
+package tcc_access
+
+import (
+	"context"
+	"testing"
+
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/stretchr/testify/require"
+)
+
+// modifies global var defined in tcc_access
+
+// TestGenerate tests the tcc_access table generation.
+func TestGenerate(t *testing.T) {
+	testContext = true
+	tccPathPrefix = "./testdata"
+	tccPathSuffix = "/test-TCC.db"
+	// test with invalid col constraint
+	_, err := Generate(context.Background(), table.QueryContext{
+		Constraints: map[string]table.ConstraintList{
+			"notacol": {
+				Affinity: table.ColumnTypeText,
+				Constraints: []table.Constraint{
+					{
+						Operator:   table.OperatorEquals,
+						Expression: "val",
+					},
+				},
+			},
+		},
+	})
+	require.Error(t, err)
```

Comment:
Why does `Generate` fail here? (I don't see any checks of , not saying we should have it, as a wrong column will probably be caught by osquery/osquery-go before the `Generate` is called.)

Reply:
So `Columns` is called to define valid columns, then osquery checks the constraints against those columns, and only _then_ is `Generate` called?